src/pyj/read_book/cfi.pyj

```diff
@@ -677,17 +661,18 @@ def scroll_to(cfi, callback, doc):  # {{{
 
             x = (point_.a*rect.left + (1-point_.a)*rect.right)
             y = (rect.top + rect.bottom)/2
-            x, y = viewport_to_document(x, y, ndoc)
+            x, y = scroll_viewport.viewport_to_document(x, y, ndoc)
             if callback:
                 callback(x, y)
     else:
         node = point_.node
         scroll_viewport.scroll_into_view(node)
-        scroll_viewport.reset_transforms()  # needed for viewport_to_document()
 
         fn = def():
             r = node.getBoundingClientRect()
-            x, y = viewport_to_document(r.left, r.top, node.ownerDocument)
+            # Start of element is right side in RTL, so be sure to get that side in RTL mode
+            x, y = scroll_viewport.viewport_to_document(
+                r.left if scroll_viewport.ltr() else r.right, r.top, node.ownerDocument)
```

Comment:
In cases where the top left of an element is used to get its start position, and we're in RTL mode, use the top right since that's where things begin in horizontal RTL mode.